.github/workflows/ios.yml

```diff
@@ -0,0 +1,24 @@
+name: iOS
+
+on:
+    push:
+    workflow_dispatch:
+    pull_request:
+      branches:
+      - main
+
+permissions: read-all
+
+jobs:
+  iOS:
+    runs-on: ${{matrix.os}}
+    strategy:
+      matrix:
+        os: [ mac-12, mac-13 ]
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      - uses: lukka/get-cmake@8be6cca406b575906541e8e3b885d46f416bba39 # v3.27.7
+      - run: python3 utils/git-sync-deps
+      - run: cmake -S . -B build -D CMAKE_SYSTEM_NAME=iOS -D SPIRV_SKIP_EXECUTABLES=ON
```

Comment:
Currently iOS builds only pass if

SPIRV_SKIP_EXECUTABLES is ON. Otherwise you get installation failures:
```
[cmake] CMake Error at tools/CMakeLists.txt:93 (install):
[cmake]   install TARGETS given no BUNDLE DESTINATION for MACOSX_BUNDLE executable
[cmake]   target "spirv-as".
[cmake] 
[cmake] 
[cmake] CMake Error at tools/CMakeLists.txt:94 (export):
[cmake]   export Export set "SPIRV-Tools-toolsTargets" not found.
```

Ideally this would be automatically handled for the user. For now however this is good enough.